### PR TITLE
Home page

### PR DESCRIPTION
**Pull request by kosodrzewina**
## The first draft of the home page layout.

- removed components left from demo flutter project
---
- added placeholder svg logo in assets/icons
- added placeholder png avatar in assets/images
- added flutter_svg to dev_dependencies
---
- created two listviews, one horizontal (recently updated) and one vertical (recently added)
- added bottom navigation bar (only visual)

image

https://user-images.githubusercontent.com/34711852/124772065-f5830100-df3b-11eb-95db-82ed550f383e.mp4

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -46,10 +47,10 @@ flutter:
   # included with your application, so that you can use the icons in
   # the material Icons class.
   uses-material-design: true
-  # To add assets to your application, add an assets section, like this:
-  # assets:
-  #   - images/a_dot_burr.jpeg
-  #   - images/a_dot_ham.jpeg
+
+  assets:
+    - assets/icons/logo.svg
```
Comment (shilangyu) [inline pubspec.yaml]: it is enough if you include the whole folder:
```
- assets/
- assets/icons/
- assets/images/
```

File: lib/styles/custom_icons.dart [hunk 2]
```diff
@@ -0,0 +1 @@
+const iconLogo = 'assets/icons/logo.svg';
```
Comment (shilangyu) [inline lib/styles/custom_icons.dart]: check out https://pub.dev/packages/flutter_gen

File: lib/main.dart [hunk 3]
```diff
@@ -9,24 +16,125 @@ class MyApp extends StatelessWidget {
...
+          child: Text(
+            text,
+          ),
+        ),
+      ),
+    ),
+  );
+}
+
+Widget renderRecentlyAddedListElement(
```
Comment (shilangyu) [inline lib/main.dart]: Create new widgets instead of helper function, these are not registered in the widget tree

File: lib/main.dart [hunk 4]
```diff
@@ -9,24 +16,125 @@ class MyApp extends StatelessWidget {
...
+  BuildContext context,
+  String text,
+  Color color,
+) {
+  return Container(
+    margin: const EdgeInsets.only(
```
Comment (shilangyu) [inline lib/main.dart]: Use the Padding widget instead

File: lib/main.dart [hunk 5]
```diff
@@ -9,24 +16,125 @@ class MyApp extends StatelessWidget {
...
+      top: 5,
+      bottom: 5,
+    ),
+    child: SizedBox(
+      height: 45,
+      child: ElevatedButton(
+        style: ButtonStyle(
```
Comment (shilangyu) [inline lib/main.dart]: You can use ButtonStyle.fromStyle

File: lib/main.dart [hunk 6]
```diff
@@ -9,24 +16,125 @@ class MyApp extends StatelessWidget {
...
+            ),
+          ),
+        ),
+        onPressed: () {},
+        child: Row(
+          children: [
+            Container(
+              decoration: BoxDecoration(
+                color: color,
+                borderRadius: const BorderRadius.all(
```
Comment (shilangyu) [inline lib/main.dart]: You can use `shape: BoxShape.circle`

File: lib/main.dart [hunk 7]
```diff
@@ -9,24 +16,125 @@ class MyApp extends StatelessWidget {
...
+                  ),
+                ),
+              ),
+              margin: const EdgeInsets.only(
+                right: 10,
+              ),
+              height: 25,
+              width: 25,
+            ),
+            Align(
```
Comment (shilangyu) [inline lib/main.dart]: use `mainAxisAlignment` on the Row instead of aligning

File: lib/main.dart [hunk 8]
```diff
@@ -46,68 +154,162 @@ class MyHomePage extends StatefulWidget {
...
+        bottom: PreferredSize(
+          preferredSize: const Size.fromHeight(0.5),
+          child: Container(
+            color: Colors.grey,
+            height: 0.5,
+          ),
+        ),
         title: Text(widget.title),
+        elevation: 0,
+        leading: Container(
```
Comment (shilangyu) [inline lib/main.dart]: Use Padding widget

File: lib/main.dart [hunk 9]
```diff
@@ -46,68 +154,162 @@ class MyHomePage extends StatefulWidget {
...
+              ),
+            ),
+            child: const CircleAvatar(
+              radius: 15,
+              backgroundImage: AssetImage(imageAvatar),
+            ),
+          ),
+        ],
       ),
       body: Center(
```
Comment (shilangyu) [inline lib/main.dart]: I don't think center does anything here

File: lib/main.dart [hunk 10]
```diff
@@ -46,68 +154,162 @@ class MyHomePage extends StatefulWidget {
...
-          // Column has various properties to control how it sizes itself and
-          // how it positions its children. Here we use mainAxisAlignment to
-          // center the children vertically; the main axis here is the vertical
-          // axis because Columns are vertical (the cross axis would be
-          // horizontal).
-          mainAxisAlignment: MainAxisAlignment.center,
-          children: <Widget>[
-            const Text(
-              'You have pushed the button this many times:',
+        child: Container(
```
Comment (shilangyu) [inline lib/main.dart]: use Padding widget

File: lib/main.dart [hunk 11]
```diff
@@ -46,68 +154,162 @@ class MyHomePage extends StatefulWidget {
...
-    // by the _incrementCounter method above.
-    //
-    // The Flutter framework has been optimized to make rerunning build methods
-    // fast, so that you can just rebuild anything that needs updating rather
-    // than having to individually change instances of widgets.
     return Scaffold(
       appBar: AppBar(
-        // Here we take the value from the MyHomePage object that was created by
-        // the App.build method, and use it to set our appbar title.
+        brightness: Brightness.dark,
```
Comment (shilangyu) [inline lib/main.dart]: Why force dark mode on the app bar?

File: lib/main.dart [hunk 12]
```diff
@@ -9,24 +16,125 @@ class MyApp extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
-      title: 'Flutter Demo',
+      title: 'Playlister',
       theme: ThemeData(
```
Comment (shilangyu) [inline lib/main.dart]: Move theme to a separate file (it will get big fast)

File: lib/main.dart [hunk 13]
```diff
@@ -46,68 +154,162 @@ class MyHomePage extends StatefulWidget {
...
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+      bottomNavigationBar: BottomNavigationBar(
+        backgroundColor: Colors.grey[900],
+        selectedItemColor: Colors.red,
+        unselectedItemColor: Colors.white,
```
Comment (shilangyu) [inline lib/main.dart]: Move parameters to theme

File: lib/main.dart [hunk 14]
```diff
@@ -9,24 +16,125 @@ class MyApp extends StatelessWidget {
...
+        child: Align(
+          alignment: Alignment.bottomLeft,
+          child: Text(
+            text,
+          ),
+        ),
+      ),
+    ),
+  );
+}
```
Comment (shilangyu) [inline lib/main.dart]: Same as below

**Review by shilangyu — CHANGES_REQUESTED** [review]

File: lib/main.dart (hunk 11 above)
Comment (kosodrzewina) [reply lib/main.dart:161]: currently the app is in dark mode, notification bar icons are black by default, so they wouldn't be visible in light mode

File: lib/main.dart [hunk 15]
```diff
@@ -9,20 +15,111 @@ class MyApp extends StatelessWidget {
...
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      width: 100,
+      margin: const EdgeInsets.symmetric(
+        horizontal: 10,
+      ),
+      child: ElevatedButton(
```
Comment (shilangyu) [inline lib/main.dart]: This tile isnt semantically a button, but rather a surface that is clickable. Use `InkWell` instead

File: lib/main.dart [hunk 16]
```diff
@@ -9,20 +15,111 @@ class MyApp extends StatelessWidget {
...
+          ),
+          onPressed: () {},
+          child: Row(
+            children: [
+              Container(
+                decoration: BoxDecoration(
+                  color: color,
+                  shape: BoxShape.circle,
+                ),
+                margin: const EdgeInsets.only(
```
Comment (shilangyu) [inline lib/main.dart]: Put a `SizedBox(width: 10)` between the container and the text instead of using margin

File: lib/main.dart [hunk 17]
```diff
@@ -46,68 +143,145 @@ class MyHomePage extends StatefulWidget {
...
       appBar: AppBar(
-        // Here we take the value from the MyHomePage object that was created by
-        // the App.build method, and use it to set our appbar title.
+        bottom: PreferredSize(
+          preferredSize: const Size.fromHeight(0.5),
+          child: Container(
+            color: Colors.grey,
+            height: 0.5,
+          ),
+        ),
```
Comment (shilangyu) [inline lib/main.dart]: You can apply the Border to the `shape:` property of `AppBar`, would be less hacky

File: lib/main.dart [hunk 18]
```diff
@@ -46,68 +143,145 @@ class MyHomePage extends StatefulWidget {
...
+          children: [
+            const Text(
+              'Recently updated',
+            ),
+            Container(
+              margin: const EdgeInsets.symmetric(
+                vertical: 10,
+              ),
+              height: 100,
+              child: ListView(
```
Comment (shilangyu) [inline lib/main.dart]: ListView accepts padding (can be used instead of using margin above)

File: lib/main.dart [hunk 19]
```diff
@@ -46,68 +143,145 @@ class MyHomePage extends StatefulWidget {
...
             ),
-            Text(
-              '$_counter',
-              style: Theme.of(context).textTheme.headline4,
+            Expanded(
+              child: Padding(
+                padding: const EdgeInsets.symmetric(
+                  horizontal: 10,
+                ),
+                child: ListView(
```
Comment (shilangyu) [inline lib/main.dart]: Same here, padding can be moved to ListView

**Review by shilangyu — CHANGES_REQUESTED** [review]

**Review by shilangyu — APPROVED** [review]

**Pull request merged**